lib/bitcoin/SidetreeTransactionData.ts

```diff
@@ -0,0 +1,131 @@
+
+    if (!writer) {
+      console.info(`Valid sidetree data was found but no valid writer was found for transaction id: ${bitcoinTransaction.id}`);
+      return undefined;
+    }
+
+    return new SidetreeTransactionData(sidetreeData, writer);
+  }
+
+  private static getValidSidetreeDataFromOutputs (transactionId: string, transactionOutputs: BitcoinOutputModel[], sidetreePrefix: string): string | undefined {
+
+    let sidetreeDataToReturn: string | undefined = undefined;
+
+    for (let i = 0; i < transactionOutputs.length; i++) {
+      const currentOutput = transactionOutputs[i];
+      const sidetreeDataForThisOutput = SidetreeTransactionData.getSidetreeDataFromOutputIfExist(currentOutput, sidetreePrefix);
+
+      if (sidetreeDataForThisOutput) {
+
+        const oneSidetreeDataAlreadyFound = sidetreeDataToReturn !== undefined;
+
+        if (oneSidetreeDataAlreadyFound) {
+          console.info(`More than one sidetree transactions were found in the outputs of transaction id: ${transactionId}`);
+          return undefined;
+        }
+
+        sidetreeDataToReturn = sidetreeDataForThisOutput;
+      }
+    }
+
+    return sidetreeDataToReturn;
+  }
+
+  private static getSidetreeDataFromOutputIfExist (transactionOutput: BitcoinOutputModel, sidetreePrefix: string): string | undefined {
+
+    // check for returned data for sidetree prefix
+    const hexDataMatches = transactionOutput.scriptAsmAsString.match(/\s*OP_RETURN ([0-9a-fA-F]+)$/);
+
+    if (hexDataMatches && hexDataMatches.length !== 0) {
+
+      const data = Buffer.from(hexDataMatches[1], 'hex').toString();
+
+      if (data.startsWith(sidetreePrefix)) {
+        return data.slice(sidetreePrefix.length);
+      }
+    }
+
+    // Nothing was found
+    return undefined;
+  }
+
+  private static getValidWriterFromInputs (transactionId: string, transactionInputs: BitcoinInputModel[]): string | undefined {
+
+    const validPublickey = SidetreeTransactionData.getValidPublicKeyFromInputs(transactionId, transactionInputs);
+
+    if (!validPublickey) {
+      return undefined;
+    }
+
+    // This library uses the public-key-hash-output format of the public key as the writer so
+    // convert the public key into the expected foramat.
+    const publicKeyAsBuffer = Buffer.from(validPublickey, 'hex');
+    const publicKeyHashBuffer = crypto.Hash.sha256ripemd160(publicKeyAsBuffer);
+
+    return publicKeyHashBuffer.toString('hex');
+  }
+
+  private static getValidPublicKeyFromInputs (transactionId: string, transactionInputs: BitcoinInputModel[]): string | undefined {
+
+    // First get all the public keys from the inputs
+    const allPublicKeys = transactionInputs.map(input => {
+      const scriptAsmParts = input.scriptAsmAsString.split(' ');
+
+      // If the publickey is not present then just use 'undefined'
+      return scriptAsmParts.length >= 2 ? scriptAsmParts[1] : undefined;

```

Comment:
For assurance, are you able to add a comment explaining why [1] is always the public key when length is great than 1?

Reply:
Added a comment to figure out this assumption in the next PR.